test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+	if err != nil {
+		log.Error("Error create Apicast Route", err)
+		return "", err
+	}
+	routeHost := route.Spec.Host
+	log.Info("routeHost: " + routeHost)
+	return routeHost, nil
+}
+
+func customerLogin() error {
+	log.Info("Customer Login")
+	token, err := getCustomerAminPasswordTokenFromTerm("Enter Customer Admin user Token, and press Enter: ")
+	if err != nil {
+		log.Error("Error login customer, can't get token: ", err)
+		return err
+	}
+	command := "oc login --token=" + token
+	err = run_shell_command(command)
+	if err != nil {
+		log.Error("Error login customer: ", err)
+		return err
+	}
+	return nil
+}
+
+func getCustomerAminPasswordTokenFromTerm(prompt string) (string, error) {
+	fmt.Print(prompt)
+	bytepw, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return "", err
+	}
+	pass := string(bytepw)
+	return pass, nil
+}
+
+func run_shell_command(command string) error {
+	log.Info(command)
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		log.Error("Error: ", err)
+		return err
+	}
+	return nil
+}
+
+// Configure the service to use the self-managed APIcast instead of the built-in APIcast for API.
+// See https://access.redhat.com/documentation/en-us/red_hat_openshift_api_management/1/topic/a702e803-bbc8-47af-91a4-e73befd3da00
+// step 11
+func promoteSelfManagedAPIcastInUI() (string, error) {
+	message := "This is the manual step - Configure the service to use the self-managed APIcast instead of the built-in APIcast for API. \n"
+	message += "a. Navigate to 3scale Admin Portal. You can use the following command to find route  \"oc get routes --namespace redhat-rhoam-3scale | grep admin\" \n"
+	message += "b. In the Products section, click API → Integration → Settings → APIcast Self Managed.\n"
+	message += "c. Change the Staging Public Base URL. Replace api-3scale-apicast- with selfmanaged-. \n"
+	message += "d. Click Update Product. \n"
+	message += "e. Click API → Configuration → Promote to Staging \n"
+	message += "f. Copy user_key value (from Staging APIcast - Example curl for testing) to command prompt \n"
+	log.Info(message)
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Waiting for manual step completion. Copy user_key value here and Press enter when done :")
+	userKey, err := reader.ReadString('\n')
+	if err != nil {
+		log.Error("Error: ", err)
+		return "", err
+	}
+	if len(userKey) > 0 {
+		userKey = userKey[:len(userKey)-1]
+	}
+	log.Info("user_key: " + userKey)
+	log.Info("Promote - completed, continue ...")
+	return userKey, err
+}
+
+func promoteSelfManagedAPIcast(ctx context.Context, client k8sclient.Client, promoteManually bool) (string, error) {
+	userKey := ""
+	var err error
+	if promoteManually {
+		userKey, err = promoteSelfManagedAPIcastInUI()
+		if err != nil {
+			log.Error("Error promote Self-managed APIcast. Exit...", err)
+			return "", err
+		}
+	} else {
+		//TODO - WIP - for full automation
```

Comment:
The plan - to integrate the script into Pipeline, so there are some minor comments in script (for future).

Reply:
Yes, that's fine to complete it in the future. Could you add a link to the follow on JIRA here in the comment where it will get addressed so it's easier for someone to find that JIRA in the future.

Reply:
Done

Reply:
Done. links to Jira added, like: ///TODO - will be done in full automation task, https://issues.redhat.com/browse/MGDAPI-3037